src/components/FiltersList.tsx

```diff
@@ -1,50 +1,45 @@
-import { GristLabelType, TableRowType } from '@common/types/gristData'
+import { TableRowType } from '@common/types/gristData'
 import classNames from '@lib/classNames'
 import { getLabelRenderer } from '@lib/getLabelRenderer'
+import { useUrlState } from '@lib/UrlStateContext'
 import { useUserGeolocation } from '@lib/hooks/useUserGeolocation'
 import { useLabels } from '@lib/LabelsContext'
 import { useTexts } from '@lib/TextsContext'
-import { useRouter } from 'next/router'
-import { FC, useState } from 'react'
-import { PrimaryButton } from './PrimaryButton'
+import { FC } from 'react'
 import { SwitchButton } from './SwitchButton'
-import { TextLink } from './TextLink'
 
 export const FiltersList: FC<{
   recordsWithOnlyLabels: TableRowType['fields']['Schlagworte'][]
-}> = ({ recordsWithOnlyLabels }) => {
+}> = () => {
   const texts = useTexts()
   const labels = useLabels()
-  const [activeFilters, setActiveFilters] = useState<GristLabelType[]>([])
-  const { push, query } = useRouter()
-  const {
-    useGeolocation,
-    setGeolocationUsage,
-    geolocationIsForbidden,
-    latitude,
-    longitude,
-  } = useUserGeolocation()
-  const filteredRecords = recordsWithOnlyLabels.filter((r) =>
-    activeFilters.every((f) => r.find((id) => id === f.id))
-  )
+  const [urlState, updateUrlState] = useUrlState()
+  const tags = urlState.tags || []
+  const { useGeolocation, setGeolocationUsage, geolocationIsForbidden } =
+    useUserGeolocation()
+
   const group1 = labels.filter(({ fields }) => fields.group === 'gruppe-1')
   const group2 = labels.filter(({ fields }) => fields.group === 'gruppe-2')
   const group3 = labels.filter(({ fields }) => fields.group === 'gruppe-3')
   const targetAudience = labels.filter(
     ({ fields }) => fields.group === 'zielpublikum'
   )
-  const someTargetFiltersActive = targetAudience.some((f) =>
-    activeFilters.find((l) => l.fields.key === f.fields.key)
+  const someTargetFiltersActive = targetAudience.some((l) =>
+    tags.find((f) => f === l.id)
```

Comment:
I would suggest we also use more descriptive variables here. Perhaps we make `targetAudience` plural, e.g. `targetGroups`? Then we first have `targetGroup` instead of `l` and `tag` instead of `f`. It's not super important and feel free to do otherwise, but I think that reasoning will get easier with different names.

Reply:
Very good point! I renamed the variables!